Sources/StreamChatUI/Navigation/ChatChannelListRouter.swift

```diff
@@ -28,13 +28,14 @@ open class ChatChannelListRouter: NavigationRouter<ChatChannelListVC>, Component
             for: cid,
             channelListQuery: rootViewController.controller.query
         )
-
-        guard let navController = rootNavigationController else {
-            log.error("Can't push chat detail, no navigation controller available")
-            return
-        }
         
-        navController.show(vc, sender: self)
+        if let splitVC = rootViewController.splitViewController {
+            splitVC.showDetailViewController(UINavigationController(rootViewController: vc), sender: self)
+        } else if let navigationVC = rootViewController.navigationController {
+            navigationVC.show(vc, sender: self)
+        } else {
+            rootViewController.show(UINavigationController(rootViewController: vc), sender: self)
+        }
```

Comment:
@evsaev Why are we making this change as part of this PR? I'm afraid this might conflict with some customers' logic.

Reply:
Another idea: If we are doing this, would it makes sense to set the `navigationController.transitionDelegate` to our own `StreamModalTransitioningDelegate`? (Check `DemoChatChannelListRouter`). We have a custom modal transition delegate to fix the Inverted message list

Reply:
> I'm not sure if we want to bring this logic to the SDK tho

With this, we cover all the scenarios of showing the channel list (in navigation/split/on its own) and execute correct navigation when the channel is opened.

Previously, if the channel list was not embedded into navigation, nothing was happening when channel is selected. We also saw how complex it was with split vc.

So for me it feels like a good change.

Reply:
> I'm afraid this might conflict with some customers' logic.

If customers have custom presentation logic (other then push), they currently have no other option but to override without calling `super`. So we are safe

Reply:
> Another idea: If we are doing this, would it makes sense to set the navigationController.transitionDelegate to our own StreamModalTransitioningDelegate? (Check DemoChatChannelListRouter). We have a custom modal transition delegate to fix the Inverted message list modal issue

Nice one, will add 👍

Reply:
@evsaev The problem is that if they dig into the code and see the `split` implementation, they will think that the split is supported out of the box, and it is not **yet**. Currently, that Split implementation was added to the DemoApp to debug stuff only

Reply:
Yeah, but we can say the same about the navigation controller. As of now, we require `ChatChannelListVC` to be embedded into the navigation controller **by the integrator**, otherwise, the channel is not opened. We explain that in docs and probably we'll need a section for split

Reply:
> The Split Implementation is out of the box and should be out of the scope of this PR. We are already changing too many things in this PR already. IMO we should keep the Split implementation a DemoApp only feature for now.

Demo is an integration example where we put ourselves into customers' shoes. In this case, refactoring has highlighted the complexity of working with split-view and other issues. Changes introduced in the router make it much simpler (before, it felt like fighting against the SDK).

If we refer to the current implementation, I actually think it needs an improvement 🙂 Like, why do we make an assumption that the channel list is embedded into navigation and freak out when it's not? We either leave `showChannel` empty and leave it to the customer to implement or handle all the common use-cases (split, navigation, none of it)

@GetStream/ios-developers what's your opinion?